### ISSUE TITLE
How to run the ALVR  server with debug  ?

### ISSUE THREAD

**Issue by Hope10086** [alvr-org/ALVR#1284]
I need to get the ` session_log.txt` to learn how it work.  I try to open the` ALVR Launcher.exe` afther command  `cargo xtask build-server` but the steamVR would go into meltdown .   I also tried typing the following command in the terminal :`"cargo run -p alvr_launcher "`  the I got  this hint `DEBUG [ureq::stream] connecting to 127.0.0.1:8082 at 127.0.0.1:8082` . It seems some errors  occur between alvr_launcher and steamVR.

**Comment by zarik5** [comment]
You can enable the log in the settings (Extra tab).

**Comment by Hope10086** [comment]
Thanks ! I also to know if the   steamVR-diver ‘s log will be printf into the session_log.txt

**Comment by zarik5** [comment]
If I interpreted correctly your question, yes it should be logged in session_log.txt, but only log generated by the ALVR driver. Log generated by SteamVR is stored elsewhere..

**Comment by Hope10086** [comment]
Thanks!
I noticed that you  comment  out the code about defining "`ALVR_DEBUG_LOG`" in the `path：alvr/ server/ build.rs `.Then the function `"_log"` `in the server/cpp/alvr_server/ logger.cpp  `will invoke functions in the  driverlog.cpp .
But the function "Debug" in the` vulkan-layer /util logger.cpp ` 
wouldn't take effect because we didn't define "
`ALVR_DEBUG_LOG`
It 's right?

**Comment by Hope10086** [comment]
Is it not possible to output logging  at the debug level in a` release` version server？I choose  “Debug” on the Notification level  
but I only get [Info] log in the` session_log .txt`
![image](https://user-images.githubusercontent.com/112925968/204730349-62a119c8-8ed3-42b7-9d3c-840631c4f681.png)

I try to run the  Debug version server but the steamVR will  block  the GameSupport  & alvr_server again and again.


![image](https://user-images.githubusercontent.com/112925968/204731445-0a43c23b-a582-4e48-8b2f-bfdbd1385449.png)

**Comment by zarik5** [comment]
Debug log is disabled in release mode. You have to compile without `--release`, but that makes SteamVR crash for me. I still don't understand why, this is a long standing bug.

**Comment by Hope10086** [comment]
I want to know if  there are  other members in your team  could compile and run in Debug mode . Is there a requirement for computer hardware?

**Comment by Hope10086** [comment]
> Debug log is disabled in release mode. You have to compile without `--release`, but that makes SteamVR crash for me. I still don't understand why, this is a long standing bug.



I took the approach of copying the file  `session.json`  to the path  `ALVR\build\alvr_server_windows` to get some logs in  the Debug mode. And the session_log.txt is following:
[session_log.txt](https://github.com/alvr-org/ALVR/files/10137840/session_log.txt)

It seems some errors ocurr in the Vulkan Initialization.

The other thing  is that  I  have tried to add a  extra  Debug  function created by myself  before  executing  `_log(format, args, LogDebug);` .The   I commented  out   `_log`   and  defined  macro  `ALVR_DEBUG_LOG`   for Debug  to work .  I got a log.txt which  have some differet from your` session_log.txt` 


[steamVRlog11-26-15.txt](https://github.com/alvr-org/ALVR/files/10137950/steamVRlog11-26-15.txt)
It shows that VSync keep   sleeping .   I know  the VSync  is related to GPU rendering and screen display.
Now that I think about it, this is a sign of a  Vulkan rendering problem. 
And I also try to change  some `Debug("XXXXXXX")` into ` Info ("XXXXXX")`  . The result is that  it would get  `log.txt`   normally when the number  of changing is  small but  it makes  SteamVR crash when  the number  of changing is  large .

Maybe  the invoking of  `Debug("XXXXXXX")`   incurs a large delay in the main thread that causes  other threads wrong then resulting in Vulkan Initialization  failed?
Of course   it is just my guess  .After all, I'm not familiar with how ALVR works
